Dockerfile

```diff
@@ -1,24 +1,40 @@
 FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:8.0 AS build
 ARG TARGETARCH
-WORKDIR /source
+WORKDIR /src
 
-# Copy files
-COPY . .
-WORKDIR /source/main/GarnetServer
+# Copy csproj and restore as distinct layers
+COPY libs/client/*.csproj libs/client/
+COPY libs/cluster/*.csproj libs/cluster/
+COPY libs/common/*.csproj libs/common/
+COPY libs/host/*.csproj libs/host/
+COPY libs/server/*.csproj libs/server/
+COPY libs/storage/Tsavorite/cs/src/core/*.csproj libs/storage/Tsavorite/cs/src/core/
+COPY libs/storage/Tsavorite/cs/src/devices/AzureStorageDevice/*.csproj libs/storage/Tsavorite/cs/src/devices/AzureStorageDevice/
+COPY main/GarnetServer/*.csproj main/GarnetServer/
+COPY metrics/HdrHistogram/*.csproj metrics/HdrHistogram/
 
-RUN dotnet restore -a $TARGETARCH
-RUN dotnet build -a $TARGETARCH -c Release
+RUN dotnet restore main/GarnetServer/GarnetServer.csproj -a $TARGETARCH
 
-# Copy and publish app and libraries
-RUN dotnet publish -a $TARGETARCH -c Release -o /app --self-contained false -f net8.0
+# Copy everthing else and publish app
+COPY Garnet.snk Garnet.snk
+COPY libs/ libs/
+COPY main/ main/
+COPY metrics/ metrics/
+COPY test/testcerts test/testcerts
+
+WORKDIR /src/main/GarnetServer
+RUN dotnet publish -a $TARGETARCH -c Release -o /app --no-restore --self-contained false -f net8.0
 
 # Final stage/image
-FROM mcr.microsoft.com/dotnet/runtime:8.0
+FROM mcr.microsoft.com/dotnet/runtime:8.0 AS runtime
 WORKDIR /app
 COPY --from=build /app .
 
+# Run container as a non-root user
+USER $APP_UID
```

Comment:
Where is `APP_UID` coming from? It's not a standard build environment variable AFAIK, and it's not imported with `ARG`>

Reply:
It's coming from [base](https://github.com/dotnet/dotnet-docker/blob/main/src/runtime-deps/8.0/bookworm-slim/amd64/Dockerfile) .NET image as default.